File: grind-75/cpp/266. Invert Binary Tree.cpp
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
    public:
        TreeNode* swapBranches(TreeNode* node) {
            if (!node) return node;

            auto tmp = node->left;
            node->left = node->right;
            node->right = tmp;
            
            swapBranches(node->left);
            swapBranches(node->right);

            return node;
        }

        TreeNode* invertTree(TreeNode* root) {
            return this->swapBranches(root);
        }
};

File: grind-75/python/266. Invert Binary Tree.py
# Definition for a binary tree node.
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        def invertBranches(node: TreeNode):
            if not node: return node
            tmp = node.right
            node.right = node.left
            node.left = tmp

            invertBranches(node.right)
            invertBranches(node.left)

            return node

        return invertBranches(root)